src/Microsoft.AspNet.Mvc.Core/Formatters/DelegatingEnumerable.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.AspNet.Mvc
+{
+    /// <summary>
+    /// Helper class to serialize <see cref="IEnumerable{T}"/> types by delegating them through a concrete implementation."/>.
+    /// </summary>
+    /// <typeparam name="T">The interface implementing <see cref="IEnumerable{T}"/> to proxy.</typeparam>
+    public sealed class DelegatingEnumerable<T> : IEnumerable<T>
+    {
+        private IEnumerable<object> _source;
+        private readonly IWrapperProvider _wrapperProvider;
+        private readonly Type _wrappedEnumerableElementType;
+
+        /// <summary>
+        /// Initialize a DelegatingEnumerable. This constructor is necessary for 
+        /// <see cref="System.Runtime.Serialization.DataContractSerializer"/> to work.
```

Comment:
yeah, most of the comments in this file were a copy-paste from Web API...will change them...